pilot/cmd/pilot-agent/main.go

```diff
@@ -489,3 +500,37 @@ func main() {
 		os.Exit(-1)
 	}
 }
+
+// GetClusterDomainName returns cluster domain name. For efficiency, it runs just once
+// all subsequent calls will just return domain name value.
+func GetClusterDomainName() string {
+	once.Do(func() {
+		f, err := os.Open(resolverFileName)
+		if err == nil {
+			defer f.Close()
+			domainName = getClusterDomainName(f)
+
+		} else {
+			domainName = defaultDomainName
+		}
+	})
+
+	return domainName
+}
+
+func getClusterDomainName(r io.Reader) string {
+	scanner := bufio.NewScanner(r)
+	for scanner.Scan() {
+		elements := strings.Split(scanner.Text(), " ")
+		if elements[0] != "search" {
```

Comment:
Please add a comment explaining what you're pulling out of the resolve config file, something like:
```go
// Kubernetes uses the cluster's local name in the resolve.conf's search list;
// the first entry is the cluster's fully qualified name, including the prefix `svc.`, like:
//     search <namespace>.svc.<cluster name> svc.<cluster name> <cluster name>
// for example, a pod in the Istio namespace with a default k8s install has:
//     search istio-system.svc.cluster.local svc.cluster.local cluster.local
// We look for the one with the `svc` prefix and grab the cluster name from it.

Reply:
Will do, thanks for suggestion.